DOC add warning note regarding data leakage by using quantile_transform

NeilBotelho opened the pull request:
Added a warning to docstring of quantile_transform to use
QuantileTransformer instead


#### Reference Issues/PRs
Addresses issue #17402
adds data leakage warning to quantile_transform in preprocessing/_data.py 

In collaboration with @asubramaniyan 
#DataUmbrella

NeilBotelho committed:
Add warning to quantile_transform

Added a warning to docstring of quantile_transform to use
QuantileTransformer instead

glemaitre commented on sklearn/preprocessing/_data.py:
```diff
@@ -2810,6 +2810,19 @@ def quantile_transform(X, *, axis=0, n_quantiles=1000,
+        In general, we recommend using
+        :class:`~sklearn.preprocessing.QuantileTransformer` within a
+        :ref:`Pipeline <pipeline>` in order to prevent most risks of data
+        leaking:`pipe=make_pipeline(QuantileTransformer(),LogisticRegression)`.
```
```suggestion
        leaking:`pipe=make_pipeline(QuantileTransformer(), LogisticRegression())`.
```

NeilBotelho replied:
I'll make the changes but I'll also add a line break  to conform to flake8 as otherwise the line becomes 83 characters
```suggestion
        leaking:`pipe=make_pipeline(QuantileTransformer(),
        LogisticRegression)`.
```

glemaitre reviewed (commented):
Otherwise LGTM

NeilBotelho committed:
Update sklearn/preprocessing/_data.py

Add brackets to LogisticRegression function and add line break to warning to make it conform to flake8

glemaitre commented:
Thank you @NeilBotelho
